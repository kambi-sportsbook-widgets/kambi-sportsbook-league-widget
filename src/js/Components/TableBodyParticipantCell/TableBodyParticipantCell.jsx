import React from 'react';
import PropTypes from 'prop-types';
import { translationModule } from 'kambi-widget-core-library';
import styles from './TableBodyParticipantCell.scss';

const t = translationModule.getTranslation.bind(translationModule);

const TableBodyParticipantCell = ({ name }) => {
   return (
      <td className={styles.general} title={t('Participant name')}>{name}</td>
   );
};

TableBodyParticipantCell.propTypes = {
   /**
    * Participant name
    */
   name: PropTypes.string.isRequired
};

export default TableBodyParticipantCell;
