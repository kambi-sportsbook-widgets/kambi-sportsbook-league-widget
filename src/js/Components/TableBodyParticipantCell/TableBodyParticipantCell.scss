@import '../TableBodyShared';

.general {
   @extend .cell;

   font-weight: bold;
   padding-right: 6px;
   line-height: 18px;
}