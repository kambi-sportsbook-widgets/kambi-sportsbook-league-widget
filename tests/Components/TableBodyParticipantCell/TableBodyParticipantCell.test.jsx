/* eslint-env jest */
import React, { Children } from 'react';
import TableBodyParticipantCell from '../../../src/js/Components/TableBodyParticipantCell/TableBodyParticipantCell';
import ReactTestRenderer from 'react-test-renderer';

jest.mock('kambi-widget-core-library', () => ({
   translationModule: {
      getTranslation: (key) => `Translated: "${key}"`
   }
}));

describe('TableBodyParticipantCell DOM rendering', () => {

   it('renders correctly', () => {
      const tree = ReactTestRenderer.create(
         <TableBodyParticipantCell name="Antoni" />
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
