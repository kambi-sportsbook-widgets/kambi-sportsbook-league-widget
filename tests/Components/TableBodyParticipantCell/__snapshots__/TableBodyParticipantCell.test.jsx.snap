// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyParticipantCell DOM rendering renders correctly 1`] = `
<td
  className="general"
  title="Translated: \\"Participant name\\""
>
  Antoni
</td>
`;
